### PR TITLE
add error message for `-fno-llvm` `-flld`

### PR DESCRIPTION
**Pull request by Rexicon226**
cc @andrewrk @kubkon

### CODE REVIEW

**Comment by andrewrk** [comment]
Can you please explain the purpose of this change in the commit message?

**Comment by Rexicon226** [comment]
How's this?

**Comment by andrewrk** [comment]
> Zig will never support linking in stage2 with LLD as the self-hosted backends are linker driven.

I don't think that's accurate. Since zig supports `build-obj`, that functionality can be repurposed to pass the output to LLD. Not sure where you got this idea from.

**Comment by Rexicon226** [comment]
Ah my apologies, I did not phrase that correctly. One moment.

File: src/Compilation/Config.zig [hunk 1]
```diff
@@ -119,6 +119,7 @@ pub const ResolveError = error{
     ZigLacksTargetSupport,
     EmittingBinaryRequiresLlvmLibrary,
     LldIncompatibleObjectFormat,
+    LldIncompatibleWithStage2,
```
Comment (kubkon) [inline src/Compilation/Config.zig:122]: Why `Stage2`? What does it even mean? What if we described exactly what is happening, namely `LldCannotIncrementallyLink` or something like that? I mean, this is effectively what happens here, isn't it?

**Review by kubkon — CHANGES_REQUESTED** [review]

File: src/Compilation/Config.zig (hunk 1 above)
Comment (Rexicon226) [reply src/Compilation/Config.zig:122]: That's a perfect name thanks!

**Review by kubkon — APPROVED** [review]

**Pull request merged**